api/routers/report.py

```diff
@@ -102,37 +111,107 @@ async def update_reports(old_user_id: int, new_user_id: int, token: str):
 
 
 @router.post("/v1/report", tags=["Report"])
-async def insert_report(token: str, detections: List[detection]):
+async def insert_report(request : Request,
+    detections: List[detections],
+    manual_detect: int = Query(0, ge=0, le=1),
+    ):
     '''
-        Work in progress 
-        Insert report.
+        Inserts detections to the plugin database.
     '''
-    await verify_token(token, verification='verify_ban', route='[POST]/v1/report/')
+    client_host = request.client.host
+    success = JSONResponse(status_code=status.HTTP_200_OK, content=jsonable_encoder({'OK': 'OK'}))
+    bad_data = JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content=jsonable_encoder({'error': 'There was an error with your request. Please contact plugin support.'}))
+
+    # remove duplicates
+    df = pd.DataFrame([d.dict() for d in detections])
+    df.drop_duplicates(
+        subset=['reporter', 'reported', 'region_id'], inplace=True)
+
+    # data validation, there can only be one reporter, and it is unrealistic to send more then 5k reports.
+    sender = df["reporter"].values[0]
+    if len(df) > int(report_maximum) or df["reporter"].nunique() > 1:
+        logger.warning(f'IP: {client_host} | Sender {sender} | Too Many Reports or Multiple Reporters!')
+        return bad_data
+
+    # data validation, checks for correct timing
+    now = int(time.time())
+    df_time = df.ts
+    mask = (df_time > int(now + int(front_time_buffer))) | (df_time < int(now - int(back_time_buffer)))
+    if len(df_time[mask].values) > 0:
+        logger.warning(f'IP: {client_host} | Sender {sender} | Data contains out of bounds time!')
+        return bad_data
```

Comment:
filter it out don't immediately return.
then check if df is empty if empty then return the response code

Reply:
Kk